pkg/webhook/validator.go

```diff
@@ -881,3 +823,104 @@ func getNamespace(ctx context.Context, namespace string) string {
 	}
 	return namespace
 }
+
+// validateContainer will validate the container image, and any errors will use
+// field & index to craft the meaningful error message.
+// field is necessary because higher level resources come here from different
+// contexts and the container could be nested at different levels in the
+// resource
+// index is the number in the containers array from the said context.
+//
+// Returns any encountered errors, or nil in two cases:
+// All the matched policies were validated, or
+// no matching policies were found, but the PolicyControllerConfig has been
+// configured to allow images not matching any policies.
+func (v *Validator) validateContainer(ctx context.Context, c corev1.Container, namespace, field string, index int, kind, apiVersion string, labels map[string]string, ociRemoteOpts ...ociremote.Option) *apis.FieldError {
+	ref, err := name.ParseReference(c.Image)
```

Comment:
nit: one potential improvement could be avoiding to parse the reference twice, as it was done in `refOrFieldError(`.

Reply:
Yeah, thought about that, but figured that adding one more field to the signature vs. just parsing it again was lesser of two evils :)